module_utils/ansible_nailgun_cement.py

```diff
@@ -40,6 +40,30 @@ def create_server(server_url, auth, verify_ssl):
     ).save()
 
 
+# Prerequisite: create_server
+def ping_server(module):
+    try:
+        return entities.Ping().search_json()
+    except Exception as e:
+        module.fail_json(msg="Failed to connect to Foreman server: %s " % e)
+
```

Comment:
I plan on going through at some point and cleaning up the existing modules to use the shared logic.